### PR TITLE
Allow to trigger on events other than `push` and secure the webhook

### LINKED ISSUE #9: Secure the webhook
If the webhook URL leaks, anyone will be able to trigger the bot and rate limit/ban the bot.

References:
* https://docs.github.com/en/webhooks-and-events/webhooks/securing-your-webhooks
* https://github.com/wtfiscrq/github_webhooks (https://deno.land/x/github_webhooks)

### PR DESCRIPTION
**Pull request by yardenshoham**
This makes the bot more flexible and responsive. We'll run backports only on `push` events to `main`.

Label maintenance and merge queue maintenance will be not only triggered only on `push` events but also on pull requests label events. To make sure we don't do too much unnecessary work, we only trigger these when either a `reviewed` or a `backport` label has been applied.

A new environment variable is introduced, `BACKPORTER_GITHUB_SECRET`. It'll be used to verify the requests are sent from GitHub.

- Closes #9 
- Part 1 of #38 (this change is required for that)

An octokit dependency is introduced for webhooks, we can later use the octokit SDK to query GitHub.

### CODE REVIEW

**Comment by yardenshoham** [comment]
@techknowlogick please change the webhook trigger from only `push` to `push` and `pull_request`:
image

I considered sending everything but that just seems like a waste of resources.

Also, please set the `BACKPORTER_GITHUB_SECRET` environment variable to the secret you set in the webhook settings page.

**Comment by techknowlogick** [comment]
@yardenshoham I've added the additional settings (push & pr events, and the secret) to the webhook settings, would you like me to add them to your personal webhook too?

**Comment by yardenshoham** [comment]
No need, I don't use my personal webhook anymore. You can remove it

**Comment by yardenshoham** [comment]
We should land this PR ASAP as we currently trigger the backporter in every pull_request event (a lot!). @techknowlogick @jolheiser  please review

**Comment by yardenshoham** [comment]
@techknowlogick revert the `pull_request` trigger until this PR lands
image

File: README.md
Comment (delvh) [inline README.md:51]:
```suggestion
webhook on `/trigger` to run this bot.
```

**Review by delvh — APPROVED** [review]

**Comment by yardenshoham** [comment]
Gitea mergers should really get write access here so their review will count

File: src/webhook.ts
Comment (silverwind) [inline src/webhook.ts:48]: Does `req.url` contain query string in Deno? If yes, this could fail.

Comment (yardenshoham) [reply src/webhook.ts:48]: You're right, but that's fine because I don't want it to trigger if there's a query string

Comment (silverwind) [reply src/webhook.ts:48]: Might want to add a comment for it.

Comment (yardenshoham) [reply src/webhook.ts:52]: Done

**Review by jolheiser — APPROVED** [review]

**Comment by yardenshoham** [comment]
Thanks for the merge, also check https://github.com/GiteaBot/gitea-backporter/pull/41 for **full protection**

**Pull request merged**
